ibm/service/database/resource_ibm_database.go

```diff
@@ -2748,7 +2690,28 @@ func expandGroups(_groups []interface{}) []*Group {
 	return groups
 }
 
-func checkV5Groups(_ context.Context, diff *schema.ResourceDiff, meta interface{}) (err error) {
+func validateGroupScaling(groupId string, resourceName string, value int, resource *GroupResource, nodeCount int) error {
+	if nodeCount == 0 {
+		nodeCount = 1
+	}
+	if resource.StepSize == 0 {
+		return fmt.Errorf("%s group must have members scaled > 0 before scaling %s", groupId, resourceName)
+	}
+	if value < resource.Minimum/nodeCount || value > resource.Maximum/nodeCount || value%(resource.StepSize/nodeCount) != 0 {
+		if !(value == 0 && resource.IsOptional) {
+			return fmt.Errorf("%s group %s must be >= %d and <= %d in increments of %d", groupId, resourceName, resource.Minimum/nodeCount, resource.Maximum/nodeCount, resource.StepSize/nodeCount)
+		}
+	}
+	if value != resource.Allocation/nodeCount && !resource.IsAdjustable {
```

Comment:
We may want to add comments on top of each of these if statements. It's a little difficult to see why these are needed without having a strong understanding of the way we manage scaling.